bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/inst/FeatureGeometry.scala

```diff
@@ -1,143 +1,107 @@
-    val armAngle  = armAdjustment.angle
-    val guideStar = armAdjustment.guideStarCoords
-    val armTrans  = AffineTransform.getRotateInstance(armAngle.toRadians, guideStar.getX, guideStar.getY)
-    armTrans.concatenate(AffineTransform.getTranslateInstance(guideStar.getX, guideStar.getY))
-    shapes.map { armTrans.createTransformedShape }
-  }
+    case class AreaCalc(start: (Double, Double), prev: (Double, Double), sum: Double) {
+      def addPoint(next: (Double, Double)): AreaCalc = {
+        val (x0, y0) = prev
+        val (x1, y1) = next
+        AreaCalc(start, next, sum + x0 * y1 - y0 * x1)
+      }
 
-  /**
-   * Given a shape representing a guide probe arm or segment, an angle for the probe arm, and the position of the guide
-   * star, execute transformations on the shape to get the adjusted probe arm.
-   * @param shape          the shape to transform
-   * @param armAdjustment  the adjustment information for the probe arm, consisting of an angle and guide star location
-   * @return               the transformed shapes
-   */
-  def transformProbeArmForContext(shape: Shape, armAdjustment: ArmAdjustment): Shape =
-    transformProbeArmForContext(List(shape), armAdjustment).head
-
-  /**
-   * Given a list of geometry shapes, transform them as needed for display on the screen.
-   * @param shapes          the list of shapes to transform
-   * @param pixelsPerArcsec the pixel density per arcsec on the current display
-   * @param flipRA          a scaling in the y-axis: should be either 1.0 or -1.0, and not sure if this is ever -1.0
-   * @param screenPos       the final position on the screen where the adjusted guide probe arm should be placed
-   * @return                the transformed shape
-   */
-  def transformProbeArmForScreen(shapes: List[Shape], pixelsPerArcsec: Double, flipRA: Double, screenPos: Point2D): List[Shape] = {
-    val trans = AffineTransform.getTranslateInstance(screenPos.getX, screenPos.getY)
-    trans.scale(pixelsPerArcsec, pixelsPerArcsec)
-    shapes.map { trans.createTransformedShape }
-  }
+      def close: Double =
+        if (prev == start) (sum/2.0).abs
+        else addPoint(start).close
+    }
 
-  /**
-   * Given a shape, transform it as needed for display on the screen.
-   * @param shape           the shape to transform
-   * @param pixelsPerArcsec the pixel density per arcsec on the current display
-   * @param flipRA          a scaling in the y-axis: should be either 1.0 or -1.0, and not sure if this is ever -1.0
-   * @param screenPos       the final position on the screen where the adjusted guide probe arm should be placed
-   * @return                the transformed shape
-   */
-  def transformProbeArmForScreen(shape: Shape, pixelsPerArcsec: Double, flipRA: Double, screenPos: Point2D): Shape =
-    transformProbeArmForScreen(List(shape), pixelsPerArcsec, flipRA, screenPos).head
+    def go(area: Double, cur: Option[AreaCalc]): Double =
```

Comment:
Can you add `@tailrec` here just for futureproofing? It's easy for someone to come in and change a pattern-match to a fold and break tail recursion inadvertently.